packages/next/src/build/webpack/plugins/next-drop-client-page-plugin.ts

```diff
@@ -8,9 +9,36 @@ const PLUGIN_NAME = 'DropAmpFirstPagesPlugin'
 
 // Prevents outputting client pages when they are not needed
 export class DropClientPage implements webpack.WebpackPluginInstance {
-  ampPages = new Set()
+  ampPages = new Set<string>()
 
   apply(compiler: webpack.Compiler) {
+    const isRspack = !!(compiler as any).rspack
+
+    if (isRspack) {
+      // For Rspack, use a simpler emit hook-based approach
+      compiler.hooks.emit.tap(PLUGIN_NAME, (compilation: any) => {
+        Object.keys(compilation.assets).forEach((assetKey) => {
+          const asset = compilation.assets[assetKey]
+
+          if (asset?._value?.includes?.('__NEXT_DROP_CLIENT_FILE__')) {
```

Comment:
```suggestion
          if (asset?._value?.includes?.(STRING_LITERAL_DROP_BUNDLE)) {
```
nit we can still use the constant